lib/travis/build/appliances/git_wire_protocol_v2.rb

```diff
@@ -0,0 +1,22 @@
+require 'travis/build/appliances/base'
+
+module Travis
+  module Build
+    module Appliances
+      class GitWireProtocolV2 < Base
+
+        GIT_VERSION = /(\d)\.(\d+)\.(\d+)/
+        GIT_WIRE_PROTOCOL_V2_MAJOR = 2
+        GIT_WIRE_PROTOCOL_V2_MINOR = 18
+        
+        def apply
+          git_version = `git version`
+          major, minor, build = git_version.match(GIT_VERSION).captures
+          if major.to_i => GIT_WIRE_PROTOCOL_V2_MAJOR && minor.to_i => GIT_WIRE_PROTOCOL_V2_MINOR
+            sh.cmd "git config --global protocol.version 2", assert: false, echo: true
```

Comment:
I think you can always apply this as it won't do anything for versions under 2.18

Reply:
Nice! Updated!